src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,843 @@
+			'cc' => []];
+
+		$signed = LDSignature::sign($data, $owner);
+
+		logger('Deliver profile deletion for user ' . $uid . ' to ' . $inbox .' via ActivityPub', LOGGER_DEBUG);
+		HTTPSignature::transmit($signed, $inbox, $uid);
+	}
+
+	/**
+	 * @brief Transmits a profile change to a given inbox
+	 *
+	 * @param integer $uid User ID
+	 * @param string $inbox Target inbox
+	 */
+	public static function transmitProfileUpdate($uid, $inbox)
+	{
+		$owner = User::getOwnerDataById($uid);
+		$profile = APContact::getByURL($owner['url']);
+
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => 'Update',
+			'actor' => $owner['url'],
+			'object' => self::profile($uid),
+			'published' => DateTimeFormat::utcNow(DateTimeFormat::ATOM),
+			'to' => [$profile['followers']],
+			'cc' => []];
+
+		$signed = LDSignature::sign($data, $owner);
+
+		logger('Deliver profile update for user ' . $uid . ' to ' . $inbox .' via ActivityPub', LOGGER_DEBUG);
+		HTTPSignature::transmit($signed, $inbox, $uid);
+	}
+
+	/**
+	 * @brief Transmits a given activity to a target
+	 *
+	 * @param array $activity
+	 * @param string $target Target profile
+	 * @param integer $uid User ID
+	 */
+	public static function transmitActivity($activity, $target, $uid)
+	{
+		$profile = APContact::getByURL($target);
+
+		$owner = User::getOwnerDataById($uid);
+
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => $activity,
+			'actor' => $owner['url'],
+			'object' => $profile['url'],
+			'to' => $profile['url']];
+
+		logger('Sending activity ' . $activity . ' to ' . $target . ' for user ' . $uid, LOGGER_DEBUG);
+
+		$signed = LDSignature::sign($data, $owner);
+		HTTPSignature::transmit($signed, $profile['inbox'], $uid);
+	}
+
+	/**
+	 * @brief Transmit a message that the contact request had been accepted
+	 *
+	 * @param string $target Target profile
+	 * @param $id
+	 * @param integer $uid User ID
+	 */
+	public static function transmitContactAccept($target, $id, $uid)
+	{
+		$profile = APContact::getByURL($target);
+
+		$owner = User::getOwnerDataById($uid);
+		$data = ['@context' => 'https://www.w3.org/ns/activitystreams',
+			'id' => System::baseUrl() . '/activity/' . System::createGUID(),
+			'type' => 'Accept',
+			'actor' => $owner['url'],
+			'object' => ['id' => $id, 'type' => 'Follow',
+				'actor' => $profile['url'],
+				'object' => $owner['url']],
+			'to' => $profile['url']];
+
+		logger('Sending accept to ' . $target . ' for user ' . $uid . ' with id ' . $id, LOGGER_DEBUG);
+
+		$signed = LDSignature::sign($data, $owner);
+		HTTPSignature::transmit($signed, $profile['inbox'], $uid);
+	}
+
+	/**
+	 * @brief 
+	 *
+	 * @param string $target Target profile
+	 * @param $id
+	 * @param integer $uid User ID
+	 */
+	public static function transmitContactReject($target, $id, $uid)
```

Comment:
Please rename to `sendContactReject` to avoid repeating "transmit" when calling the method.